### PR TITLE
feat: extract android UserLocation FPS

### PR DESCRIPTION
**Pull request by marekmeissner**
## Description

Added `androidPreferredFramesPerSecond` prop for UserLocation that allows to set max FPS for user location animation. This allows to improve performance especially on low-end android devices. 

Change is based on research of maplibre-native `LocationComponent.java`, which serves and encourages to use `setMaximumFps` method for this feature purpose.

## Checklist

- [x] I have tested this on a device/simulator for each compatible OS
- [X] I formatted JS and TS files with running `yarn lint:fix` in the root folder
- [X] I have run tests via `yarn test` in the root folder
- [X] I updated the documentation with running `yarn generate` in the root folder
- [X] I mentioned this change in `CHANGELOG.md`
- [n/a?] I updated the typings files (`index.d.ts`)
- [X] I added/updated a sample (`/example`)

## Screenshot OR Video

https://github.com/user-attachments/assets/e077855c-486e-4fcd-a805-6fc2fa635aeb

### CODE REVIEW

File: android/rctmln/src/main/java/com/maplibre/rctmln/components/location/LocationComponentManager.java [hunk 1]
```diff
@@ -67,6 +67,15 @@ public void setRenderMode(@RenderMode.Mode int renderMode) {
         }
     }
 
+    public void setPreferredFramesPerSecond(int preferredFramesPerSecond) {
+       if(preferredFramesPerSecond <= 0) {
```
Comment (caspg) [inline android/rctmln/src/main/java/com/maplibre/rctmln/components/location/LocationComponentManager.java:71]: Should we also check for a max value?

**Review by caspg — APPROVED** [review]
Cool stuff. Looking good 👍

File: android/rctmln/src/main/java/com/maplibre/rctmln/components/location/LocationComponentManager.java (hunk 1 above)
Comment (marekmeissner) [reply android/rctmln/src/main/java/com/maplibre/rctmln/components/location/LocationComponentManager.java:71]: There's no max value for this setting in Mapbox, by default it is set to 3-15 (depending on zoom). IMO we can leave it like that. What do you think?

Comment (caspg) [reply android/rctmln/src/main/java/com/maplibre/rctmln/components/location/LocationComponentManager.java:71]: Yup, makes sense 👍

**Comment by marekmeissner** [comment]
@caspg I updated CHANGELOG in the latest commit to match rules

**Comment by tyrauber** [comment]
@marekmeissner, thank you for this.  Useful feature, and the docs and test coverage are appreciated. Will go out in 10.0.0-alpha.12.

**Pull request merged**